AndelaOSP/Andela-Socials
# #160617749 Setup client testing

#### What Does This PR Do?

Setup e2e and unit testing and add some smoke tests

#### Any Background Context You Want To Provide?

We basically want to reach a stage where we have a stable app that we have confidence in.
Testing it is the beginning of that vision.

#### How can this be manually tested?

Ensure you have the recent changes

- `cd client/`
- `yarn`
- `yarn test`
- Ensure server is started `scripts/start.sh`
- `yarn test:visual`

#### What are the relevant pivotal tracker stories?

[160617749](https://www.pivotaltracker.com/n/projects/2174978/stories/160617749)

#### Screenshot(s)

<img width="414" alt="screen shot 2018-10-16 at 15 14 26" src="https://user-images.githubusercontent.com/19430799/47015673-48964380-d156-11e8-889d-960d71faffa8.png">

<img width="433" alt="screen shot 2018-10-16 at 15 14 46" src="https://user-images.githubusercontent.com/19430799/47015683-4df38e00-d156-11e8-8320-ec5107bc1441.png">
